components/ratgdo/switch/ratgdo_switch.cpp

```diff
@@ -21,6 +22,18 @@ namespace ratgdo {
             this->parent_->subscribe_learn_state([=](LearnState state) {
                 this->publish_state(state == LearnState::ACTIVE);
             });
+        } else if (this->switch_type_ == SwitchType::RATGDO_EMULATION) {
+            bool value = false;
+            this->pref_ = global_preferences->make_preference<float>(this->get_object_id_hash());
+            this->pref_.load(&value);
```

Comment:
since esp8266 stores prefs in flash this may cause all prefs to be lost when a new one is added since they are restored in order